### ISSUE TITLE
An apology and change in direction.

### ISSUE THREAD

**Issue by andrewmd5** [Codeusa/Borderless-Gaming#417]
Building and creating software has always been my passion. I left school at 15 to do it full time and since then, have been improving myself as a developer and still seeking out new challenges. Back in 2014, I created Borderless Gaming and was fortunate to have it become widely popular thanks to the support of communities on Reddit and other internet circles. The influx of contributors the project received was nothing short of amazing, and the praise that was received was a welcomed change from past projects.

The trust the community placed in me to create utilities like Borderless Gaming was something I didn't quite grasp a few years ago; after all most teenagers are happy to get 15 minutes of fame. However, in the past two years, I have fractured that trust severely. From the GPL -> MPL incident to the recent ad modal controversy, I've time and time again fumbled and in some way and undermined the trust given to me by users of my software. I can no longer ignore the deep-seated feelings in the community some hold.

When these events occurred the vitriol I was reading was bewildering. Ignorance was attributed to malice, my behavior becoming fuel for the fire. I dug in; I refused to hear other perspectives, I shut down conversations and wiped my nose at the outrage before finally reversing course. The damage was done, permeant marks on my reputation and only I am too blame. I am 23 now and to be honest; I'm quite embarrassed looking back on all this.

Through my unprofessional behavior, I've not only let the users of this and my other projects down but the many contributors over time as well. I was unable to admit I had screwed up on multiple occasions and forgot the most fundamental thing about being a software developer, remembering the user. Ultimately, by being unable to separate my ego from the product, I acted in a way that doesn't align with the person I want to be.

Being able to take a step back and reflect on one's actions can be increasingly uncomfortable, especially when confronted with ugly truths; however, it is necessary to grow as an individual. So to the community, I am genuinely sorry. I will actively work to improve myself and set the bar high in what I create.

I would like to step away as the primary maintainer of this project. Partly because of my past actions which have damaged trust between users and myself and on the other hand I just no longer have time these days.  I feel the project would be better off in someone else's hands so I can focus on my company and being a better person. 

If you are interested, please email me: hi@andrew.im

**Comment by Uriei** [comment]
I don't know about everyone else, but I'm more than happy with how this software is handled, yes, people make mistakes, but if they can be fixed, and afterward are indeed fixed, I see no problem with keeping it this way, mistakes is how we truly learn, you can learn many things, but you can also forget them, but if it was from a mistake? Woah, I doubt you'll forget that one, specially when it seems that affected you so hard to get to this point.

TL;DR: I have no issues at all, everyone makes mistakes, I'm happy how this is going and I wouldn't want it to change.

**Comment by s-thom** [comment]
I can't remember how I found Borderless Gaming 3 years ago, but I'm so glad I did. It even ended up being one of my [first contributions](https://github.com/Codeusa/Borderless-Gaming/issues/159) to open source software, because I wanted a way to match the different versions of Minecraft so I didn't need to add a new favourite for each version.

While I haven't been active at all in this project since then, I have been keeping an eye on the project, including when the dramas were happening. I'm glad that you've been able to take a step back, realise what happened, and have taken steps to fix the mistakes that happened. I know that it's not easy to own your past actions, and I think what you've said is difficult to say, and I respect you for that.

I hope this project goes to a maintainer that cares for the project, and is willing to take it to the next level, wherever that may be.

**Comment by PointyDev** [comment]
This software has helped so much over the time that I've been using it (2 years now?), and it's kinda sad to see the primary maintainer have to deal with controversy like this. I hope you go on to better things mate, keep doing what you do.

**Comment by DevelDoe** [comment]
Don't resign, you learned from your mistakes, hard to find people like that

**Comment by Rab-C** [comment]
Thanks for the hard work! It's a really helpful little app you've brought into the world. Drama comes with youth - Hell, 23 is still very young in the greater scheme of things- and it's not like people were paying you & entered into a contract for your time & effort. So although self analysis, criticism & improvement are always great, don't run yourself down too far, mate! You gave away part of your life for free...

**Comment by KarlOfDuty** [comment]
I also agree, I was surprised at the rainway advert back in the day, but I have not exactly lost any confidence in you.

**Comment by shani-ace** [comment]
I didn't even know about the connection between Rainway and Borderless Gaming - I actually knew about Rainway long before discovering this lovely tool - and never heard of any of the aforementioned issues.

So while I don't see the need to step down because of those issues - Borderless Gaming is still a great tool, after all - it feels more like you're stepping down to focus more on Rainway.

**Comment by twistedsanity** [comment]
To be honest, I wasn't around to know the history, and can only go on the present. Finding someone who admits they have done wrong in the past and is striving to do better is a wonderful attribute, and I truly commend you for it. Thanks for a great project which makes my gaming experience better. Either way I hope this project flourishes as there is nothing quite like it and I would struggle without it.

**Comment by DailytheNoob** [comment]
I've been using this software for a few years now and have no clue what is going on, but the software has been working great every time

**Comment by OMG-Abaddon** [comment]
This has been the greatest tool I've ever had. I never understood why full-fledged gfx engines couldn't handle a god damn borderless fullscreen setting for years, which this app seemed to handle flawlessly so long ago as an external app.

I've been using it since it was beta IIRC, it sometimes malfunctioned back then, but soon became my go-to tool for windowed game fixing.

I don't know if I'm thinking about the same thing as you when you say you "made mistakes", for me the ones I know were pretty much minor and a fair decision... it's easy to backseat while someone else has to make an influential decision. It might not have been the smoothest way to pull it off, but it wasn't the end of the world.

Whatever happens in the end, thank you for this awesome software.

**Comment by draeath** [comment]
Any trust with me you've lost, you've just regained.

I would not be unhappy if you continued maintaining it yourself.

**Comment by m-alice** [comment]
no need to blame yourself. being an individual is good. reputation is nothing, remember the bridge builder.
 just be yourself.

**Comment by Santiagodraco** [comment]
I've been using BG for quite some time.  I appreciate the work put into it and find it a valuable tool.

That said I am of two minds here with this thread.  First if you don't have time to support it any longer, or simply don't want to, that is perfectly ok.   Life happens.  Of course it would be good to keep it working should something happen but that's up to you.

On the other hand saying you are walking away from the project because of "past actions" seems... like not taking responsibility for your actions and simply running away to make it easier on yourself.   If you really feel you've done something wrong to others the mature thing to do is to run towards the problem and make up for those past actions.  By committing more, not less, and showing you care and can change.

Frankly I don't have any personal knowledge of what you'd done wrong in the past or whatever... my reply here is based on your original post and not any other knowledge I have.

**Comment by PrimordialHelios** [comment]
I love your software, a lot. I've even bought Borderless Gaming on Steam because I love what you're doing and want to support you.
This apology shows incredible maturity. Sure, you screwed up at the time, but you eventually came around, realized your mistakes, and you're now apologizing for them. I have absolutely no issue with you as the maintainer of this project.
As for lack of time, that's understandable. Honestly I'm fine with you just working on this only occasionally, and I trust you more than some random taking over the project.
However, if you want to step away due to the vitriol of the userbase (because the internet can't be nice, apparently), that's completely understandable, and I'm not sure anyone would blame you for wanting to get away from that. Hell, even Van Rossum stepped down because of that kind of shit.
Regardless of your choice, thank you for your apology, your software, and your time as a maintainer :)

**Comment by chriscarver** [comment]
No, no I don't want to go to the release page for the latest update. STOP IT.

**Comment by cheatfreak47** [comment]
@chriscarver you want to be in #420
But yes @Codeusa if you're gonna drop off maintenance for Borderless Gaming, at least push the latest update to the steam build first. That update notification is getting on many people's nerves I think.

**Comment by Codeusa** [comment]
@cheatfreak47 IIRC everyone getting that message disabled Steam integration so BG thinks its the non-steam version. I will update Steam this week.

To everyone else, thank you all for the kind words. I will take them into consideration.

**Comment by Santiagodraco** [comment]
Great to hear Codeusa!

**Comment by DeathLoveCrymo** [comment]
Codeusa - even before I knew about this, I had already changed my bad review on Steam to a good one - because I had realized how much of a difference you had made in development and how useful this tool has become to me ever since those changes were made.

Don't give this product up to someone else, wear it proudly on your sleeve!

**Comment by Codeusa** [comment]
@Chaython Borderless Gaming has been on Steam for years now and the store page has remained exactly the same. Including this large paragraph right on the page.

![](https://i.imgur.com/aLFM6jh.png)

Clicking "View the manual" even takes people to GitHub. How exactly did I mislead anyone?

**Comment by Michael-Z-Freeman** [comment]
I don't know all the history behind this but your software has always been useful to me. What ever you decide I think you've done well, man ! Many developers would not even be willing to post something like this (or anyone else for that matter). No one is perfect and we're all learning.

**Comment by jons-own-stuff** [comment]
> Don't resign, you learned from your mistakes, hard to find people like that

Yeah. Do you have any idea how much dumb shit I did when I was younger? What I did doesn't matter but what I learned from it does. People forget things very quickly anyway. One presidential tweet and all of the USA is wondering what the "BG" icon is on their taskbar.

Also, this apology does a lot. Very few people would do this because it takes a lot of self-reflection that we often don't want to face (nerd alert: actually physical pain and "being wrong" are processed in the brain right next to each other and in the same way). Getting over that barrier after you've already dug in your heels shows a lot of courage.

**Comment by Gitoffthelawn** [comment]
I have never heard of this project, nor used it.  I stumbled on this issue while searching GitHub.

I want to quickly say how impressed I am with the fine person who wrote the original post, and the kind replies that have followed.  It makes me genuinely happy and feel good.

We would all like to never make mistakes, but when we do, I believe the key is to acknowledge them, apologize if appropriate, and no matter what, work towards not repeating them.

**Comment by 17Orion** [comment]
Everyone makes mistakes buddy! Hang in there, ignore the haters!
I continue to use Borderless Gaming on both of my gaming rigs (both paid versions on Steam)
Keep up the good work!
